python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -802,73 +819,33 @@ def readSrcMatches(self, dataRefList, dataset, repoInfo, aliasDictList=None):
-                                   "Do you have the correct astrometry_net_data setup?")
-            noMatches = False
-            if len(matches) < 8:
-                for m in matches:
-                    if not hasattr(m.first, "get"):
-                        matches = []
-                        noMatches = True
-                        break
-
-            # LSST reads in a_net catalogs with flux in "nanojanskys", so must
-            # convert to AB.
-            if not noMatches:
-                matches = matchNanojanskyToAB(matches)
-                if repoInfo.hscRun and self.config.doAddAperFluxHsc:
-                    addApertureFluxesHSC(matches, prefix="second_")
-
-            if not matches:
+            matches = loadDenormalizeAndUnpackMatches(catalog, packedMatches, refObjLoader)
+            # LSST reads in reference catalogs with flux in "nanojanskys", so
+            # must convert to AB.
+            matches = matchNanojanskyToAB(matches)
+            if matches.empty:
                 self.log.warn("No matches for {:s}".format(dataRef.dataId))
                 continue
 
-            matchMeta = repoInfo.butler.get(dataset, dataRef.dataId,
-                                            flags=afwTable.SOURCE_IO_NO_FOOTPRINTS).getTable().getMetadata()
-            catalog = matchesToCatalog(matches, matchMeta)
             if self.config.doApplyExternalSkyWcs:
                 # Update "distance" between reference and source matches based
                 # on external-calibration positions.
                 angularDist = AngularDistance("ref_coord_ra", "src_coord_ra",
                                               "ref_coord_dec", "src_coord_dec")
-                catalog["distance"] = angularDist(catalog)
+                matches["distance"] = angularDist(matches)
 
-            # Compute Focal Plane coordinates for each source if not already
-            # there.
-            if self.config.analysisMatches.doPlotFP:
-                schema = getSchema(catalog)
-                if "src_base_FPPosition_x" not in schema and "src_focalplane_x" not in schema:
-                    det = repoInfo.butler.get("calexp_detector", dataRef.dataId)
-                    catalog = addFpPoint(det, catalog, prefix="src_")
-            # Optionally backout aperture corrections
-            if self.config.doBackoutApCorr:
-                catalog = backoutApCorr(catalog)
-            # Need to set the alias map for the matched catalog sources
-            if aliasDictList:
-                catalog = setAliasMaps(catalog, aliasDictList, prefix="src_")
             # To avoid multiple counting when visit overlaps multiple tracts
             noTractId = dataRef.dataId.copy()
             noTractId.pop("tract")
             if noTractId not in dataIdSubList:
-                catList.append(catalog)
+                allMatches = matches if allMatches is None else pd.concat([allMatches, matches], axis=0)
```

Comment:
I don't see anywhere in this function where `allMatches` would be anything other than `None`.  Am I missing something?